backend/moonstreamapi/providers/moonworm_provider.py

```diff
@@ -0,0 +1,453 @@
+
+
+def python_type(expected_type: str) -> Any:
+    if expected_type.startswith("int"):
+        return int
+    elif expected_type.startswith("str"):
+        return str
+    elif expected_type == "float":
+        return float
+    elif expected_type == "bool":
+        return bool
+    elif expected_type == "tuple":
+        return list
+    else:
+        raise ValueError(f"Cannot convert to python type {expected_type}")
+
+
+class MoonwormProvider:
+    def __init__(
+        self,
+        event_type: str,
+        blockchain: AvailableBlockchainType,
+        description: str,
+        streamboaundary_range_limit: int,
+    ):
+        self.event_type = event_type
+        self.blockchain = blockchain
+        self.description = description
+        self.valid_period_seconds = streamboaundary_range_limit
+
+    def default_filters(self, subscriptions: List[BugoutResource]) -> Filters:
+        """
+        Default filter strings for the given list of subscriptions.
+        """
+        filters = Filters()
+        for subscription in subscriptions:
+            subscription_address = cast(
+                Optional[str], subscription.resource_data.get("address")
+            )
+            if subscription_address is not None:
+
+                # How apply labels?
+                filters.addresses.append(
+                    AddressFilters(address=subscription_address, labels=[])
+                )
+            else:
+                logger.warn(
+                    f"Could not find subscription address for subscription with resource id: {subscription.id}"
+                )
+        return filters
+
+    def events(self, row: Tuple) -> data.Event:
+        """
+        Parses a result from the result set of a database query for Ethereum transactions with block timestamp
+        into an Event object.
+        """
+        (
+            block_number,
+            address,
+            transaction_hash,
+            label_data,
+            block_timestamp,
+            log_index,
+            created_at,
+        ) = row
+        return data.Event(
+            event_type=self.event_type,
+            event_timestamp=block_timestamp,
+            event_data={
+                "hash": transaction_hash,
+                "block_number": block_number,
+                "address": address,
+                "label_data": label_data,
+                "log_index": log_index,
+                "created_at": created_at,
+            },
+        )
+
+    def parse_filters(
+        self,
+        query: StreamQuery,
+        user_subscriptions: Dict[str, List[BugoutResource]],
+    ) -> Optional[Filters]:
+        """
+        Passes raw filter strings into a Filters object which is used to construct a database query
+        for ethereum transactions.
+
+        Filter syntax is:
+        - "from:<address>" - specifies that we want to include all transactions with "<address>" as a source
```

Comment:
This syntax is wrong for the Moonstream providers. Please update docstring to reflect correct syntax. This documentation is currently our best guide about how to use filters.